docs/mapping-network-events.asciidoc

```diff
@@ -0,0 +1,152 @@
+[[ecs-mapping-network-events]]
+=== Mapping Network Events
+
+Network events capture the details of one device communicating with another. The initiator is referred to as the source and the recipient as the destination. Depending on the data source, a network event can contain details of addresses, protocols, headers, and device roles.
+
+This guide describes the different field sets available for network-related events in ECS and provides direction on the ECS best practices for mapping to them.
+
+[float]
+==== Source and destination baseline
+
+When an event contains details about the sending and receiving hosts, the baseline for capturing these values will be the <<ecs-source,source>> and <<ecs-destination,destination>> fields.
+
+Some events may also indicate each host's role in the exchange: client or server. When this information is available, the <<ecs-client,client>> and <<ecs-server,server>> fields should be used _in addition to_ the `source` and `destination` fields. The fields and values mapped under `source`/`destination` should be copied under `client`/`server`.
+
+[float]
+==== Network event mapping example
+
+Below is a DNS network event. The source device (`192.168.86.222`) makes a DNS query, acting as the client and the DNS server is the destination (`192.168.86.1`).
+
+Note this event contains additional details that would populate additional fields (such as the <<ecs-dns>>) if this was a complete mapping example. These additional fields are omitted here to focus on the network details.
+
+[source,json]
```

Comment:
Would it make sense to use an ECS version of the source data?

Reply:
I chose an example event that wasn't already mapped to ECS trying to capture each steps in the exercise of mapping the event's network-related values.

Would breaking down an event that's already mapped to ECS (e.g. a packetbeat DNS event) for the example be more useful?

Reply:
Actually I think the original makes more sense now that I've thought about it.  I think the example data threw me for a loop as any request/response protocol analysis will have clear source/destination linking to client/server..

basically for anything besides unidirectional flows (meraki netflow, aws vpcflow) & broadcast requests (dhcp) copy source/destination to client/server.

for the corner cases just use source/dest.

I think I may be getting wrapped around an axle that is as much about base semantics, as it is about using the data e.g. the network siem page displays...